File: java/src/main/java/com/workshop/architecture/fitness/clean_architecture/use_case/port/inbound/ActivateMembershipUseCase.java
package com.workshop.architecture.fitness.clean_architecture.use_case.port.inbound;

import jakarta.transaction.Transactional;

public interface ActivateMembershipUseCase {
    @Transactional
    ActivateMembershipResult activateMembership(ActivateMembershipInput input);
}

File: python/src/workshop_api/fitness/clean_architecture/use_case/port/inbound/activate_membership_use_case.py
from typing import Protocol

from .activate_membership_input import ActivateMembershipInput
from .activate_membership_result import ActivateMembershipResult


class ActivateMembershipUseCase(Protocol):
    async def activate_membership(
        self,
        input_data: ActivateMembershipInput,
    ) -> ActivateMembershipResult:
        pass
